### PR TITLE
Fix: 1773 milestone calculated and adjusted fields

### PR DESCRIPTION
**Pull request by Sepehr-Sobhani**
[ZH CARD 1773](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1773)
AND
[ZH CARD 1183](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1183)

### CODE REVIEW

File: app/tests/unit/components/Form/ProjectMilestoneReportFormSummary.test.tsx [hunk 1]
```diff
@@ -185,56 +220,56 @@ describe("The Project Milestone Report Form Summary", () => {
...
-              reportDueDate: "2020-01-10T23:59:59.999-07:00",
-              reportingRequirementId: 1,
-              hasExpenses: true,
-              calculatedGrossAmount: 567,
-              calculatedNetAmount: 789,
-              calculatedHoldbackAmount: 891,
-              adjustedNetAmount: 89,
-              adjustedGrossAmount: 67,
-              adjustedHoldbackAmount: 91,
+  it("Displays diffs of the data fields that were updated and shows latest committed values", () => {
```
Comment (BCerki) [inline app/tests/unit/components/Form/ProjectMilestoneReportFormSummary.test.tsx:223]: This one and the following one are just moving variables around? (Hard to see if there's a real diff in here)

**Review by BCerki — COMMENTED** [review]
1. On the view page, if the calculated values aren't calculated yet, there's nothing (I think we should see "This field cannot be calculated due to lack of information now".
Form:
image
View: 
image

2. I couldn't get Holdback Amount This Milestone to calculate even though as far as I can tell from the tooltip, all the required info was filled out: https://www.loom.com/share/5b03056bc6e54993a23f6f7bbd1e94a9

3. This seemed totally unrelated to your PR so I made a separate ticket: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1813

4. When first creating a project, on the summary page I don't see the adjusted/calculated labels, and the calculated values that aren't calculated show up $0.00 instead of "This field cannot be calculated due to lack of information now."
image

File: app/tests/unit/components/Form/ProjectMilestoneReportFormSummary.test.tsx (hunk 1 above)
Comment (Sepehr-Sobhani) [reply app/tests/unit/components/Form/ProjectMilestoneReportFormSummary.test.tsx:223]: You're right this one is hard to follow. Just to clarify, this commit is all about moving variables around to make the test a bit more readable and close to what we do in other tests: https://github.com/bcgov/cas-cif/pull/1805/commits/113b773f0c1fc0697f7a80f7403dc9ccf5c74b60

File: app/lib/theme/ReadOnlyFieldTemplate.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ const FieldTemplate: React.FC<FieldTemplateProps> = ({
 }) => {
   return (
     <div>
-      <div className="definition-container">
+      <div className={`${uiSchema["ui:classNames"] ?? "definition-container"}`}>
         {displayLabel && (
```
Comment (BCerki) [inline app/lib/theme/ReadOnlyFieldTemplate.tsx:19]: This is a great solution, nice work! I think it could be simplified to something like:

```<div className={ uiSchema.calculatedValueFormContextProperty ? "adjustable-calculated-value-widget":"definition-container"}`}>

Then any calculated field will get the style without us having to remember to add classnames to the uiSchema

File: app/lib/theme/widgets/ReadOnlyAdjustableCalculatedValueWidget.tsx [hunk 3]
```diff
@@ -57,21 +38,19 @@ const ReadOnlyAdjustableCalculatedValueWidget: React.FC<WidgetProps> = (
           ) : (
             <em>Not added</em>
           )}
+          <style jsx>{`
+            .adjustedValue {
+              display: flex;
+              flex-basis: 100%;
+              padding-top: 0.5em;
```
Comment (BCerki) [inline app/lib/theme/widgets/ReadOnlyAdjustableCalculatedValueWidget.tsx:45]: Overall the new styling is a huge improvement so I'd be comfortable approving this as-is, but I did come across some extra spacing when the adjusted value is filled out if you want to keep tweaking:
Screenshot from 2023-07-21 15-43-34
Screenshot from 2023-07-21 15-47-21

**Review by BCerki — COMMENTED** [review]
Looking good! Styling is SOO much better now :)

File: app/lib/theme/ReadOnlyFieldTemplate.tsx (hunk 2 above)
Comment (Sepehr-Sobhani) [reply app/lib/theme/ReadOnlyFieldTemplate.tsx:19]: Great suggestion 🤩

File: app/lib/theme/widgets/ReadOnlyAdjustableCalculatedValueWidget.tsx (hunk 3 above)
Comment (Sepehr-Sobhani) [reply app/lib/theme/widgets/ReadOnlyAdjustableCalculatedValueWidget.tsx:45]: Done

**Review by BCerki — APPROVED** [review]
I saw a couple changes in the happo shots that might just be flake? (moving tasklist, space between arrows in the diffs changing)

**Comment by Sepehr-Sobhani** [comment]
> I saw a couple changes in the happo shots that might just be flake? (moving tasklist, space between arrows in the diffs changing)

Might be, going to re-run the e2e again.

**Review by BCerki — APPROVED** [review]
